src/Codeception/Subscriber/BeforeAfterTest.php

```diff
@@ -1,45 +1,66 @@
 <?php
+
+declare(strict_types=1);
+
 namespace Codeception\Subscriber;
 
 use Codeception\Event\SuiteEvent;
 use Codeception\Events;
+use PHPUnit\Framework\Test as PHPUnitTest;
+use PHPUnit\Framework\TestSuite\DataProvider;
+use PHPUnit\Util\Test as TestUtil;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use function call_user_func;
+use function get_class;
+use function is_callable;
+use function strstr;
 
 class BeforeAfterTest implements EventSubscriberInterface
 {
-    use Shared\StaticEvents;
+    use Shared\StaticEventsTrait;
 
+    /**
+     * @var array<string, string|int[]|string[]>
```

Comment:
I don't think that this signature matches `['afterClass', 100]`.
Also declaring different type of the same property in different classes doesn't look right to me.

According to [docblock in symfony/event-dispatcher](https://github.com/symfony/event-dispatcher/blob/v5.2.3/EventSubscriberInterface.php#L39-L41), events can be defined in 3 formats:

```
     *  * ['eventName' => 'methodName']
     *  * ['eventName' => ['methodName', $priority]]
     *  * ['eventName' => [['methodName1', $priority], ['methodName2']]]
```

Must $events property be public? I didn't find any code accessing it directly - only getSubscribedEvents method in StaticEventsTrait.

Reply:
> I don't think that this signature matches ['afterClass', 100].

It was the only one that psalm was happy with, it also raised doubts for me, but the other option is to simply leave it as an array. Nested `array<>` syntax is not well interpreted by some static code analyzers.

> Must $events property be public?

I don't think so. I'll change it to `protected`.

Reply:
In the other classes I left it as: `@var array<string, string>` but it should probably be the same everywhere. So I'll just leave it as you tell me.